### PR TITLE
Fix Issues Seen with Production Data

### PR DESCRIPTION
**Pull request by mcdonnnj**
# 

## 🗣 Description
This PR does several things:
* Pulls in upstream changes from [`cisagov/skeleton-python-library`](https://github.com/cisagov/skeleton-python-library).
* Adds back in the `test` workflow.
* Fixes the `build` workflow to provide a deployment ready artifact on download instead of an artifact that was a zip of the deployment file.
* Fixes the AWS deployment package generation to only include packages missing from the lambda environment.
* Fixes issues with the lambda being unable to process files added to the S3 bucket which contained spaces in the filename.
* Add support for phases in RVA IDs being denoted with either a `.` or a `-` and generating the final RV ID into a consistent format that included phase information.

## 💭 Motivation and Context
The original motivation for this PR was to fix issues seen with the lambda deployed in production. The issues were two-fold: some files would not get processed at all, and the RV IDs generated did not correctly account for the use of "phases" in the RVA ID. The problem with files resulted from files with spaces in their filenames. When the event is given to the lambda the key representing the originating file has any spaces replaced with `+` characters. The lambda then tried to access those S3 objects with the given key which did not match how S3 stored them. This was resolved by using the `urllib.parse.unquote_plus()` function. However, if a file was uploaded with both spaces and `+` characters, the lambda would be unable to correctly resolve the filename as it is stored in S3.

## 🧪 Testing
I tested this in a testing environment to ensure data that failed to process previously was processed as expected. I also tested that the `build` workflow was correctly generating a deployment artifact in line with what is generated locally.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: fdi/findings_data_import.py [hunk 1]
```diff
@@ -196,9 +198,12 @@ def import_data(
                     finding.pop(field, None)
 
             # Get RVA ID in format DDDD(.D+) from the end of the "RVA ID" field.
```
Comment (jsf9k) [inline fdi/findings_data_import.py]: I think this comment needs to be updated, since the regex has changed.

**Review by jsf9k — APPROVED** [review]
Approved, but please see my one comment about a comment that may need updating.

File: fdi/findings_data_import.py (hunk 1 above)
Comment (mcdonnnj) [reply fdi/findings_data_import.py]: Good catch, fixed in 370bfae

**Comment by mzack5020** [comment]
Looks great! Thanks @mcdonnnj

**Review by mzack5020 — APPROVED** [review]

**Review by felddy — APPROVED** [review]
Looks good!  🐃 ❄️ 

WRT the LGTM alert:

One of Alan Perlis's well-known programming epigrams (recounted in ACM SIGPLAN Notices 17(9), September, 1982) states that "If you have a procedure with 10 parameters, you probably missed some."
 
Not sure there is anything to change here.  I was going to suggest passing `args` wholesale into import_data, but you are setting defaults.  This is not then end of the world.  

Rock on!

**Pull request merged**